ci_framework/plugins/action/ci_script.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/python
+
+# Copyright Red Hat, Inc.
+# Apache License Version 2.0 (see LICENSE
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+
+DOCUMENTATION = r'''
+---
+action: ci_script
+
+short_description: Runs a string given script remotly while providing logging
+
+description:
+- Discover latest image matching a specific pattern on remote server.
+
+options:
+    output_dir:
+        description: The path in where the script will be copied
+        required: true
+        type: str
+    cmd:
+        description: The shell script content to be run
+        required: true
+        type: str
+    debug:
+        description: Enables bash tracing in the script
+        required: true
+        type: bool
+'''
+
+EXAMPLES = r'''
+- name: Run custom script
+  register: script_output
+  ci_script:
+    output_dir: "/home/zuul/ci-framework-data/artifacts"
+    cmd: |
+      mkdir /home/zuul/test-dir
+      cd /home/zuul/test-dir
+      git clone https://github.com/openstack-k8s-operators/ci-framework.git
+'''
+
+RETURN = r'''
+changed:
+   description: Always true.
+   returned: always
+   type: bool
+failed:
+   description: True if the execution failed.
+   returned: always
+   type: bool
+rc:
+   description: Script return code.
+   returned: always
+   type: int
+stderr:
+    description: STDERR output as string.
+    returned: always
+    type: str
+stderr_line:
+    description: STDERR output as lines.
+    returned: always
+    type: list[str]
+stdout:
+    description: STDOUT output as string.
+    returned: always
+    type: str
+stdout_line:
+    description: STDOUT output as lines.
+    returned: always
+    type: list[str]
+'''
+
+import glob
+import pathlib
+import re
+import uuid
+
+from ansible.plugins.action import ActionBase
+from ansible.errors import AnsibleActionFail
+
+
+TMPL_SCRIPT = '''#!/bin/bash
+%(opts)s
+exec > >(tee -ia %(logpath)s)
+%(content)s
+'''
+
+class ActionModule(ActionBase):
+
+    def __init__(self, **kwargs):
+        super(ActionModule, self).__init__(**kwargs)
+        self.__script_file_path = pathlib.Path('~/.ansible/tmp').joinpath(uuid.uuid4().hex).absolute()
+        self.__script_file_path.parent.mkdir(parents=True, exist_ok=True)
+
+    @staticmethod
+    def __build_options(task_args):
+        opts = 'set -euo pipefail'
```

Comment:
I'd hard-code that one in the template directly. Would avoid the `\n` as shown on L102